Homework 16

annavladimab opened the pull request:
Hi, I've added hw 16 

Thank you in advance!

annavladimab committed:
feat: hw 15

annavladimab committed:
feat: hw 14

annavladimab committed:
feat: hw 16

AnastasiiaGor commented on Homeworks/homework_16/MyListener.py:
```diff
@@ -0,0 +1,26 @@
+from robot.libraries.BuiltIn import BuiltIn
+
+class MyListener:
```
Listener is expected to have end_keyword method that checks keywords' tags and makes screenshot if there is a 'screenshot' tag in the list of tags, other listener methods are not needed in this case

annavladimab replied:
MyListener has been updated
[Tags]    Screenshot has been added

AnastasiiaGor commented on Homeworks/homework_16/base_suite.robot:
```diff
@@ -0,0 +1,22 @@
+*** Test Cases ***
+User Registration and Login Test
+    Open Browser and Log In
+    [Teardown]    Close Browser
```
Could be moved to suite Setting section as it's the same for both tests in the suite
```
*** Settings ***
Test Teardown    Close Browser
```

annavladimab replied:
Test Teardown    Close Browser has been added to Settings
Thank you!

AnastasiiaGor commented on Homeworks/homework_16/base_suite.robot:
```diff
@@ -0,0 +1,22 @@
+    [Teardown]    Close Browser
+
+Product Category Test
+    Open Browser and Log In
```
`[Setup]  Open Browser and Log In`
would be better because login is considered as a precondition for this case

annavladimab replied:
[Setup]  Open Browser has been added

AnastasiiaGor commented on Homeworks/homework_16/common_keywords.resource:
```diff
@@ -0,0 +1,63 @@
+    Login Test
+
+Open Login Modal
+    Sleep    20s
```
It would be better avoid using Sleep (especially such a long sleep) 
If it's not possible to wait for a specific event before test check in some cases it would be better to leave explanatory comments

annavladimab replied:
I deleted all the Sleep parameters except for a few that before the alert, the pop-up window does not have time to appear, the page is still loading after clicking add product
and the next one when loading the page Wait Until Location Contains https://www.demoblaze.com/cart.html without it the system does not have time to check the added products

AnastasiiaGor commented on Homeworks/homework_16/base_suite.robot:
```diff
@@ -0,0 +1,22 @@
+
+*** Test Cases ***
+User Registration and Login Test
+    Open Browser and Log In
```
`Open Browser` part of this keyword is not a part of this test scenario, it would be better to move it to Setup

annavladimab replied:
I did the following: I separated the browser loading and login
and then created Common Suite in keywords which include browser loading and login as pre-req
(I don’t know how correct this is), but I didn’t find any other solutions
Thanks for your comments, it's really helpful!

annavladimab committed:
update: hw 16
[Setup]   Open Browser Test has been added
[Tags]    Screenshot has been added
Sleep has been deleted

annavladimab committed:
update: hw 16
[Setup]   Open Browser Test has been added
[Tags]    Screenshot has been added
Sleep has been deleted
